src/main/kotlin/com/example/iotmanagementdevice/mapper/DeviceMapper.kt

```diff
@@ -1,21 +1,23 @@
 package com.example.iotmanagementdevice.mapper
 
 import com.example.iotmanagementdevice.dto.device.request.DeviceCreateRequestDto
-import com.example.iotmanagementdevice.dto.device.response.DeviceResponseDto
 import com.example.iotmanagementdevice.dto.device.request.DeviceUpdateRequestDto
-import com.example.iotmanagementdevice.model.Device
+import com.example.iotmanagementdevice.dto.device.response.DeviceResponseDto
+import com.example.iotmanagementdevice.model.MongoDevice
 import org.mapstruct.InjectionStrategy
 import org.mapstruct.Mapper
 import org.mapstruct.NullValueCheckStrategy
 
-@Mapper(componentModel = "spring",
+@Mapper(
+    componentModel = "spring",
     injectionStrategy = InjectionStrategy.CONSTRUCTOR,
     nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
-    implementationPackage = "<PACKAGE_NAME>.impl")
+    implementationPackage = "<PACKAGE_NAME>.impl"
+)
 interface DeviceMapper {
-    fun toDto(device: Device): DeviceResponseDto
+    fun toDto(device: MongoDevice?): DeviceResponseDto
```

Comment:
why did you make incoming parameter nullable?

Reply:
Fixed